Source/JavaScriptCore/inspector/InjectedScriptBase.h

```diff
@@ -49,34 +49,37 @@ namespace Inspector {
 
 using AsyncCallCallback = WTF::Function<void(Protocol::ErrorString&, RefPtr<Protocol::Runtime::RemoteObject>&&, std::optional<bool>&&, std::optional<int>&&)>;
 
+JS_EXPORT_PRIVATE RefPtr<JSON::Value> toInspectorValue(JSC::JSGlobalObject*, JSC::JSValue);
+
 class JS_EXPORT_PRIVATE InjectedScriptBase {
 public:
     virtual ~InjectedScriptBase();
 
     const String& name() const { return m_name; }
-    bool hasNoValue() const { return m_injectedScriptObject.hasNoValue(); }
-    JSC::JSGlobalObject* globalObject() const { return m_injectedScriptObject.globalObject(); }
+    bool hasNoValue() const { return !m_injectedScriptObject.get(); }
+    JSC::JSGlobalObject* globalObject() const { return m_globalObject; }
 
 protected:
     InjectedScriptBase(const String& name);
-    InjectedScriptBase(const String& name, Deprecated::ScriptObject, InspectorEnvironment*);
+    InjectedScriptBase(const String& name, JSC::JSGlobalObject*, JSC::JSObject*, InspectorEnvironment*);
 
     InspectorEnvironment* inspectorEnvironment() const { return m_environment; }
 
     bool hasAccessToInspectedScriptState() const;
 
-    const Deprecated::ScriptObject& injectedScriptObject() const;
-    Expected<JSC::JSValue, NakedPtr<JSC::Exception>> callFunctionWithEvalEnabled(Deprecated::ScriptFunctionCall&) const;
-    Ref<JSON::Value> makeCall(Deprecated::ScriptFunctionCall&);
-    void makeEvalCall(Protocol::ErrorString&, Deprecated::ScriptFunctionCall&, RefPtr<Protocol::Runtime::RemoteObject>& resultObject, std::optional<bool>& wasThrown, std::optional<int>& savedResultIndex);
-    void makeAsyncCall(Deprecated::ScriptFunctionCall&, AsyncCallCallback&&);
+    JSC::JSObject* injectedScriptObject() const;
+    Expected<JSC::JSValue, NakedPtr<JSC::Exception>> callFunctionWithEvalEnabled(ScriptFunctionCall&) const;
+    Ref<JSON::Value> makeCall(ScriptFunctionCall&);
+    void makeEvalCall(Protocol::ErrorString&, ScriptFunctionCall&, RefPtr<Protocol::Runtime::RemoteObject>& resultObject, std::optional<bool>& wasThrown, std::optional<int>& savedResultIndex);
+    void makeAsyncCall(ScriptFunctionCall&, AsyncCallCallback&&);
 
 private:
     void checkCallResult(Protocol::ErrorString&, RefPtr<JSON::Value> result, RefPtr<Protocol::Runtime::RemoteObject>& resultObject, std::optional<bool>& wasThrown, std::optional<int>& savedResultIndex);
     void checkAsyncCallResult(RefPtr<JSON::Value> result, const AsyncCallCallback&);
 
     String m_name;
-    Deprecated::ScriptObject m_injectedScriptObject;
+    JSC::JSGlobalObject* m_globalObject { nullptr };
```

Comment:
Is it OK to have this as a raw pointer? Do we need to protect from garbage collection?

Reply:
Yes, it is actually OK since this is held by JSGlobalObject